### PR TITLE
Preserve intentional new lines in GraphQL

### LINKED ISSUE #3251: GraphQL: Preserve intentional blank lines
I write comprehensive documentation for my GraphQL schemas, and I keep finding that having multiple commented fields becomes very hard to navigate unless there is a blank line separating them. Otherwise it's very hard to scan and determine where a field is defined.

[Example input and output on the Playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBiABAQSug2gQwCcYBLAZxgF0AKACxhgAcykB6VgWwFcySwAjQvhJQAXgDoIhAOasAJhDCtMxcjACU6cumkI4QgDYGAnunwAdKBm68wJfFAA0OwhC6N0EAGbobfe1BkzlKeMLT6vjx86IyuXnBkvND4BujUBiQA1nBmMa5yXGARUpYYCNIicPrq4ugA8mBg+ElQKSbOJDDoTTgpZBDo-Dn46FDQALR+TamM+v046VnDpTG0EDAQ0kKM4YTODlpGPDBCG3uehLmMEHBdAO5rZDl3hJ0i0mRmhDlPXRuRtnUwUIKzgADcELkvHxSMlUmBaER8GB4IRxJYYMZZlhVBQtBxGAY4BwEDBPgA5CByODOACisE6pmAlnQ6AwABVwugAJIAEU8PgOEH4ACs4CiWVo5EgebyAISWSUcrkAWQAQnyBegwjlSYz0ThfPwSNL0Oq+QqoEr0Jyct5oXYUqN8CStTr0HrMQbWa0STKAMond6Kw3Kn5BqDSbUDHg5Lwhd2MLiEa5PLVSamvSODUy+nLUeMg0MegAeLsJNJzvggYPeX1IYCJnwy2XQgAwCHWATAJo9quQg5BcrnNoJ3m80unno9b3V5XLAapL+sRyS64AGI9IQ6yMJhuhAOCTYOhYwP-uEDB45Gp3lxyLQpXrpiZnSSB57Oglvegr2QXcbpFw+CwlA65ZpuVrFrgmAZM0CQ0PQTAsOwUyCMIYiSDI8iKMoMFPGQmhEDkJ49hQUhwNaKSoq08Avgkg4cOQZCzEY7xkF+KT2HhMpQbhlBbmyNpcpAXCwIQph3J094PHw94HEQpB4tosQkBwRAkM+Jp6tCcByLU3IwNa2jePATiDFwE7rGYBh3PgxhBFoZKDCQxC0KwhaqcBe4iScxhfsJonGKBwYQdugnDN8IwSWE6DSQiZi9LiXRKa8qmvBp1IMtpuk8gZxZGV4JnOPw5mjJZKQ2XZHSOcaLluVIHkkNAXkBexEUyiocAWCFAm2l8nVaDgsWyQlCldEM0gHDqzndIRETUCEDz6ORxY6uJS2DFIpkkD47ryWoWifCMsyEPMC6GuNIgdfg7URfxYZ9SMIgxbQMnxfW+39jpvZwNNTTfHNC27MtoWrd+JA6R0O1cntimHTEw5QGdrL9ldN2dXdYWDHAFS9FAb549+QF0d430egymKjqWagIEUemOXAZYSsWJIOHW1KzHjnxNe6mLYiTnq+ZKGTxP6jAODKAAy21wKLDgYwA6uEOqXEdCODgiSIohEmmZeDh2fKpRLOPEhuViUxZQD9yu1AA4q47ifAopVdIiEI6J9J1froeP6EFkYY715r8s0-SOvAA5Rfe7pgikXB2lDOQAAbe5mifWtpBi6ZKKf6HyMpBwrSu7G9MNJXDx3zPsLhuIwwLoP0JJhHWcAGE8X682u6CBmBAeqhqweJIo9jhzFkmkzHBhx26XKJx3afFhnWeGh3edmv3GNS3iJO4NysBLRQX0AAp7XRu-uUBjVQPBDDMGwnBRAIQgiBIUiyAoSjcof3LqNa1Cf9-e5qSfELKTUuX4SCMHINxf+fFuoYE3l0beu8qIXzhF3GADg5BEAHCuV03IMqkG0oQa+iE74oSfuhV+WEP7+nJN-X+3JaEAMgEA9AIDdqJXAWQKAUC8CMLobA60u4Mhbx8N8SAhAryRmbCIbIp4BhN0+ALL84iMysWoPgAqvt0FgWNs5CgMpkHqBlAAJXFGoyMABhaAlsUSXwxsI-aJNvhElgjIqAcieyKLJqQL0koXGdTwtQSUoUABi6lURVmgHaS4HAyLoACbBau7htRYk-CE1JsxuJmNcU8O2Nd2RpMEYaMJESIj8FMNE+i8TElpgoEBWMbEMn1JgDwbJLdAkywwa0sgsDWSsk0aiP24F+lsP0TAQxsBJTGPQDkzp1ioC2OAg49AIjEFiI6UkrYNd3GeP+N45R-jNl5PtswYJJSBLhIMJEipnhLbVO+Ak45ORtkpI7k0i5Hd2m5LgPk9whTZh9P6YM7R3dgqjOhCdCZPIpmGhmXM2CfzGALKWfYuBWBVlOJ8HcKQmRdlfX2S9JR5NBaGhxYQPFGitGEGGXoqFkyNAynlrilF4plnopyagwIL1mBYxgHcKoOADkkvigOdYxd3zwA+ayFxXKyA8pYLMlucqFUrLWVqSARg2WX0+JAWAaE6zCt8aS1kmqiR2OgGQKlQydHvDpQYmFjL0DWK1RaqArK3W9xyKcOssYTroFdi8kgEIhUKKJT4-U-iL6RlMdG8CQjMWiO1PgD4mKPEErDdoQ5y8U1WpBTS21kZ7XQqMTKdkKaPXsoAL6WBAI4EAEBGDATIMgUA45kBeD6DSEAqEwDZBgLLOwkZkAnDjvW9+Hau31pEE8Ygh8hDSFUpO1u3aRRkBLGqIQfbbj+lXFLS2y6nj1rcEwcyAAmQ93axYnX0MgEAWx8A7AAI4GDrSAZSsB5YmjCMgAAHAABnrbECATx5bbBHYQMdIBvhPtvN8edKal1IE7Su+tTwGIQag7wSMRIACKXB1hwEvfWjB-Av1yB-UgM9JGn4ZCsfuJDIAxgHvrbGct-AW3Ia7VWqtQA)

IMO, the input is much nicer than the output. And my [full schema](https://github.com/exogen/graphbrainz/blob/master/docs/schema.md) is much larger, navigating it without blank lines is a nightmare.

**Expected behavior:**

I'd expect the GraphQL behavior to match the JavaScript behavior: preserve at most one blank line (but remove any at the start & end of the document).

### PR DESCRIPTION
**Pull request by duailibe**
Fix #3251

### CODE REVIEW

**Comment by duailibe** [comment]
Will do!

I should also add support to newlines inside arguments, as the example in the issue also contains them

File: tests/graphql_newline/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  ...fragmentSpread, @include(if: ["this isn't even a boolean", "wow, that's really odd",,,,,])
+}
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+query MyQuery
+  @directive(
+    arg: 5
+
+    # comment
+    arg2: 10
+  ) {
```
Comment (azz) [inline tests/graphql_newline/__snapshots__/jsfmt.spec.js.snap]: This line looks odd to me

Comment (duailibe) [reply tests/graphql_newline/__snapshots__/jsfmt.spec.js.snap]: This was the current behavior when it breaks: [playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBHArnATgTwAQCyOAiprngAIAmAllnGDDQG5wAUAOlHngDbQBzAIJYB6ALYIYAOQCGkpHg4h+UYaIlSAarN6ZlXHqvVjJsOQqUrBI09t36QhvjY1mZ8uIuXHbm2Dp6cAZQAJR4wM4AZjRwvFRcAL4gADQgEAAOTNAAzsigsjkwyDBYmGkARliyYADWcDAAyhk1NGolZXBpVBBgyFG6OV0gbUNYMAAK1QLisv2DwwBWOQAeAELVdQ2NngAybXDzvENpEOgwGecATEcnIC1YY8ggAtUZABaovKn3WG0wAHUaFQYO9kAAOAAMaQyWAgQwBbw65RA9AwdDgU1kMzmSAGx2GQ3ENGRhLaAl4cFIEHgt2GMFkFSBILBSCuaVKshovHJAGEIOJZs8oNBDml0EMACqMvJ4haJRJAA)

Comment (azz) [reply tests/graphql_newline/__snapshots__/jsfmt.spec.js.snap]: Yeah. Don't have any better ideas either. Fine to leave it.

**Pull request merged**